src/helics/core/CoreTypes.hpp

```diff
@@ -18,15 +18,15 @@ SPDX-License-Identifier: BSD-3-Clause
 
 namespace helics {
 /** enumeration of the possible federate states*/
-enum FederateStates {
-    HELICS_CREATED, /*!> state upon creation, all registration calls are allowed*/
-    HELICS_INITIALIZING,  //!< the federation has entered initialization state and initial values
-                          //!< can be published
-    HELICS_EXECUTING,  //!< the federation has entered execution state and it now advancing in time
-    HELICS_TERMINATING,  //!< the federate is in the process of shutting down
+enum class FederateStates : std::uint8_t {
+    CREATED, /*!> state upon creation, all registration calls are allowed*/
+    INITIALIZING,  //!< the federation has entered initialization state and initial values
+                   //!< can be published
+    EXECUTING,  //!< the federation has entered execution state and it now advancing in time
+    TERMINATING,  //!< the federate is in the process of shutting down
     HELICS_ERROR,  //!< the federation has encountered an error
```

Comment:
Should this be renamed to just `ERROR`?

Reply:
ERROR is a macro on some other libraries it would be problematic to have it just be ERROR,  in other places I have it listed as ERRORED,  which could be done here.  But for the time being I just left it as HELICS_ERROR.

Reply:
ERRORED would also better match the other macro names in the enum